### ISSUE TITLE
dill issues with Python 3.3 related to super affecting transitions

### ISSUE THREAD

**Issue by aleneum** [pytransitions/transitions#236]
See [travis buid](https://travis-ci.org/pytransitions/transitions/builds/241162808) (snippet below). This issue appears for python 2.7 and 3.3. As a temporary workaround I set the requirements to `dill < 0.2.7`.
```python
Traceback (most recent call last):
  File "/home/travis/build/pytransitions/transitions/tests/test_core.py", line 428, in test_pickle
    dump = pickle.dumps(m)
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/dill/dill.py", line 279, in dumps
    dump(obj, file, protocol, byref, fmode, recurse)#, strictio)
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/dill/dill.py", line 272, in dump
    pik.dump(obj)
  File "/opt/python/3.3.5/lib/python3.3/pickle.py", line 235, in dump
    self.save(obj)
  File "/opt/python/3.3.5/lib/python3.3/pickle.py", line 342, in save
    self.save_reduce(obj=obj, *rv)
  File "/opt/python/3.3.5/lib/python3.3/pickle.py", line 430, in save_reduce
    save(state)
  File "/opt/python/3.3.5/lib/python3.3/pickle.py", line 297, in save
    f(self, obj) # Call unbound method with explicit self
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/dill/dill.py", line 869, in save_module_dict
    StockPickler.save_dict(pickler, obj)
  File "/opt/python/3.3.5/lib/python3.3/pickle.py", line 633, in save_dict
    self._batch_setitems(obj.items())
  File "/opt/python/3.3.5/lib/python3.3/pickle.py", line 666, in _batch_setitems
    save(v)
  File "/opt/python/3.3.5/lib/python3.3/pickle.py", line 297, in save
    f(self, obj) # Call unbound method with explicit self
nose.proxy.RuntimeError: maximum recursion depth exceeded in comparison...
```

**Comment by aleneum** [comment]
Okay, I boiled it down to this:

```python
import dill as pickle
from functools import partial


def get_trigger(model, trigger_name, *args, **kwargs):
    func = getattr(model, trigger_name, None)
    if func:
        return func(*args, **kwargs)


class Machine(object):

    def __init__(self):
        self.trigger = partial(get_trigger, self)

m = Machine()
dump = pickle.dumps(m)
```

works for dill 0.2.6, does not work for dill 0.2.7

**Comment by aleneum** [comment]
(One) problem solved for Python 2.7 but Python 3.3 still breaks.

**Comment by aleneum** [comment]
`dill` drops Python 3.3 in version 0.2.7 since the issue requires more effort than justifiable concerning the [EOL](https://docs.python.org/devguide/#status-of-python-branches) of Python 3.3. The issue with 3.3 is related to how `super` works and there is a workaround for that (see uqfoundation/dill#236). The big question it: Should transitions `0.6` also drop Python 3.3 support?

**Comment by aleneum** [comment]
I will go ahead and try to integrate the `super` workaround of possible.

**Comment by aleneum** [comment]
Fixed in cd7bd8939c70a4671ffa591c93e7c65de2ff0b6b
